pageserver/src/walreceiver.rs

```diff
@@ -149,12 +150,12 @@ fn walreceiver_main(
         error!("No previous WAL position");
     }
 
-    // There might be some padding after the last full record, skip it.
-    //
-    // FIXME: It probably would be better to always start streaming from the beginning
-    // of the page, or the segment, so that we could check the page/segment headers
-    // too. Just for the sake of paranoia.
-    startpoint += startpoint.calc_padding(8u32);
+    // FIXME: We have to do it to handle new segment generated by pg_resetwal at compute node startup.
+    // This hack is needed only when pageserver is connected directly to compute node, not through walkeeper.
```

Comment:
I don't understand this. How does the presence of a walkeeper change this?

Reply:
WAL keepercode also belongs to nonrel_wal branch. So it "knows" that first segment is messing unlike standard walsender in Postgres.